legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/pureToSQLQuery/pureToSQLQuery.pure

```diff
@@ -4822,10 +4827,17 @@ function meta::relational::functions::pureToSqlQuery::processAggFn(v:ValueSpecif
                       pair($f.func == or_Boolean_MANY__Boolean_1_ && $singleArgument, 'boolor'),
                       pair(true, $fn)
                     ]->filter(p | $p.first)->first()->toOne().second;
-                    let aggFunc = newDynaFunction(
+
+                    let aggFunc = if ($f.func == wavg_WavgRowMapper_MANY__Float_1_,
+                    | let parameters = $f.parametersValues->map(p|$p->processAggFn($params, $state));
+                      if ($parameters->size() == 1 && $parameters->toOne()->instanceOf(DynaFunction) && $parameters->cast(@DynaFunction).parameters->size()== 2 && $parameters->cast(@DynaFunction).name == 'wavgRowMapper',
```

Comment:
When will this condition evaluate to false and does a test case cover it?

Reply:
yes its covered in the test written, that else part is called when being called by old TDS flow